extends KinematicBody2D

signal health_value
signal time_value

var velocity = Vector2(0,0)
var coins = 0
const SPEED = 250
const GRAVITY = 35
const JUMPFORCE = -900

func _physics_process(delta):
	if Input.is_action_pressed("right"):
		velocity.x = SPEED
		$Sprite.play("walk")
		$Sprite.flip_h = false
	elif Input.is_action_pressed("left"):
		velocity.x = -SPEED
		$Sprite.play("walk")
		$Sprite.flip_h = true
	else:
		$Sprite.play("idle")
		
	if not is_on_floor():
		$Sprite.play("air")
		
	velocity.y = velocity.y + GRAVITY
	
	if Input.is_action_just_pressed("jump") and is_on_floor():
		$Jump_sound.play()
		velocity.y = JUMPFORCE

	velocity = move_and_slide(velocity, Vector2.UP)
	
	velocity.x = lerp(velocity.x,0, 0.2)


func _on_Fall_Zone_body_entered(body):
	print("fall zone")
	#get_tree().reload_current_scene()
	position.x = 20
	position.y = 20
	emit_signal("health_value")
	emit_signal("time_value")
	
	
func bounce():
	velocity.y = JUMPFORCE * 0.7
	
func ouch(var enemyposx):
	set_modulate(Color(1,0.3,0.3,0.4))
	velocity.y = JUMPFORCE * 0.5
	emit_signal("health_value")
	
	if position.x < enemyposx:
		velocity.x = -800
	elif position.x > enemyposx:
		velocity.x = 800
		
	Input.action_release("left")
	Input.action_release("right")
	$Timer.start()


func _on_Timer_timeout():
	set_modulate(Color(1,1,1,1))

